packages/browser/src/core/queue/event-queue.ts

```diff
@@ -43,9 +43,10 @@ export class EventQueue extends Emitter {
   async register(
     ctx: Context,
     plugin: Plugin,
-    instance: Analytics
+    instance: Analytics,
+    pluginConfig?: PluginConfig
   ): Promise<void> {
-    await Promise.resolve(plugin.load(ctx, instance))
+    await Promise.resolve(plugin.load(ctx, instance, pluginConfig))
```

Comment:
The plugin will use this as `pluginConfig.dependencies.storage`. It kinda looks bit abrasive to me, why not just pass it in as a flattened argument. It is highly unlikely that there'll be anything more we'll need to make available to plugins.